### PR TITLE
ORC-1609: Fix the compilation problem of `TestJsonFileDump` in branch 1.9

### PR DESCRIPTION
**Pull request by cxzl25**
### Why are the changes needed?
Fix `branch_1.9` compile problem

https://github.com/apache/orc/actions/runs/7769676224/job/21189006989

### How was this patch tested?
GA

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: java/tools/src/test/org/apache/orc/tools/TestJsonFileDump.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public void testDoubleNaNAndInfinite() throws Exception {
     FileOutputStream myOut = new FileOutputStream(workDir + File.separator + outputFilename);
 
     // replace stdout and run command
-    System.setOut(new PrintStream(myOut, true, StandardCharsets.UTF_8));
+    System.setOut(new PrintStream(myOut, true, StandardCharsets.UTF_8.toString()));
```
Comment (paliwalashish) [inline java/tools/src/test/org/apache/orc/tools/TestJsonFileDump.java:162]: Looks good, does it make sense to have the same fix here and in main branch? The above API is available in both JDK 8 and 17, and the Charset one is only available in JDK17. Just a suggestion, else fix looks good

**Comment by dongjoon-hyun** [comment]
Thanks. Please use a new Jira ID.

File: java/tools/src/test/org/apache/orc/tools/TestJsonFileDump.java (hunk 1 above)
Comment (dongjoon-hyun) [reply java/tools/src/test/org/apache/orc/tools/TestJsonFileDump.java:162]: No need for main/branch-2.0 because the minimum is Java 17

**Comment by cxzl25** [comment]
> Please use a new Jira ID

Thanks. Done.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to branch-1.9.

**Pull request closed without merging**